src/main/java/org/jenkinsci/plugins/runselector/context/RunSelectorContext.java

```diff
@@ -314,19 +347,50 @@ private void copyExtensionListFrom(RunSelectorCommonContext src) {
     }
 
     /**
+     * Constructs the environment variables for the current build.
+     *
+     * @return the current build environment variables
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private EnvVars constructEnvVars() throws IOException, InterruptedException {
+        EnvVars envVars = build.getEnvironment(listener);
+        if (build instanceof AbstractBuild) {
+            envVars.overrideAll(((AbstractBuild<?, ?>) build).getBuildVariables()); // Add in matrix axes..
```

Comment:
Please use `envVars.putAll` instead, as `overrideAll` removes empty variables.
Copyartifact has a bug for that: https://issues.jenkins-ci.org/browse/JENKINS-36554